homeassistant/components/rpi_power/binary_sensor.py

```diff
@@ -0,0 +1,104 @@
+"""
+A sensor platform which detects underruns and capped status from the official Raspberry Pi Kernel.
+
+Minimal Kernel needed is 4.14+
+"""
+import logging
+import os
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PROBLEM,
+    BinarySensorEntity,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SYSFILE = "/sys/class/hwmon/hwmon0/in0_lcrit_alarm"
+SYSFILE_LEGACY = "/sys/devices/platform/soc/soc:firmware/get_throttled"
+
+UNDERVOLTAGE_STICKY_BIT = 1 << 16
+
+DESCRIPTION_NORMALIZED = "Voltage normalized. Everything is working as intended."
+DESCRIPTION_UNDER_VOLTAGE = "Under-voltage was detected. Consider getting a uninterruptible power supply for your Raspberry Pi."
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the sensor platform."""
+    if discovery_info is None:
+        return
+
+    if os.path.isfile(SYSFILE):
+        under_voltage = UnderVoltage()
+    elif os.path.isfile(SYSFILE_LEGACY):  # support older kernel
+        under_voltage = UnderVoltage(legacy=True)
+    else:
+        _LOGGER.critical(
```

Comment:
Error level is enough.
```suggestion
        _LOGGER.error(
```